source/MRViewer/MRSurfaceContoursWidget.cpp

```diff
@@ -0,0 +1,421 @@
+#include "MRSurfaceContoursWidget.h"
+#include "MRViewport.h"
+#include "MRMesh/MRHistoryAction.h"
+#include "MRAppendHistory.h"
+#include <GLFW/glfw3.h>
+#include "MRMesh/MRMesh.h"
+#include "MRMesh/MRSphereObject.h"
+#include "MRMesh/MRObjectMesh.h"
+
+namespace MR
+{
+
+void updateBaseColor( std::shared_ptr<SurfacePointWidget> point, const Color& color )
+{
+    auto params = point->getParameters();
+    params.baseColor = color;
+    point->setParameters( params );
+}
+
+// History classes;
+class AddPointActionPickerPoint : public HistoryAction
```

Comment:
history classes should be in public access for users to use them in plugins

Reply:
ok